File: src/main/java/com/harragan/battleshipsboot/controllers/PlayerController.java
package com.harragan.battleshipsboot.controllers;

import com.harragan.battleshipsboot.facades.PlayerAddingFacade;
import com.harragan.battleshipsboot.model.game.Player;
import com.harragan.battleshipsboot.repositorys.PlayerRepository;
import com.harragan.battleshipsboot.service.PlayerService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import static org.springframework.web.bind.annotation.RequestMethod.POST;

@RestController
public class PlayerController {

    @Autowired
    private PlayerAddingFacade playerAddingFacade;

    @Autowired
    private PlayerService playerService;

    @Autowired
    private PlayerRepository playerRepository;

    @RequestMapping(value = "/addplayer", method = POST)
    public ResponseEntity<Integer> createPlayer(@RequestBody PlayerRequestWrapper playerRequestWrapper) {
        int playerId = playerAddingFacade.createPlayerAndJoinToGame(playerRequestWrapper.getPlayerName()
                , playerRequestWrapper.getGameId());
        return new ResponseEntity<>(playerId, HttpStatus.CREATED);
    }

    @RequestMapping(value = "/readytostart/{playerNo}", method = POST)
    @ResponseStatus(value = HttpStatus.OK)
    public void playerReady(@PathVariable int playerNo) {
        playerService.setPlayerIsReady(playerNo, playerRepository);
    }
}

File: src/main/java/com/harragan/battleshipsboot/controllers/PlayerController.kt
package com.harragan.battleshipsboot.controllers

import com.harragan.battleshipsboot.facades.PlayerAddingFacade
import com.harragan.battleshipsboot.repositorys.PlayerRepository
import com.harragan.battleshipsboot.service.PlayerService
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
class PlayerController(
        private val playerAddingFacade: PlayerAddingFacade,
        private val playerService: PlayerService,
        private val playerRepository: PlayerRepository
) {
    @RequestMapping(value = ["/addplayer"], method = [RequestMethod.POST])
    @ResponseStatus(HttpStatus.CREATED)
    fun createPlayer(@RequestBody playerRequestWrapper: PlayerRequestWrapper): Int {
        return playerAddingFacade.createPlayerAndJoinToGame(playerRequestWrapper.playerName
                , playerRequestWrapper.gameId)
    }

    @RequestMapping(value = ["/readytostart/{playerNo}"], method = [RequestMethod.POST])
    @ResponseStatus(HttpStatus.OK)
    fun playerReady(@PathVariable playerNo: Int) {
        playerService.setPlayerIsReady(playerNo, playerRepository)
    }
}
